src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -0,0 +1,321 @@
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtils;
+
+/**
+ * This class is used internally in the build process to write a property file
+ * with short descriptions (the first sentences) of TokenTypes constants.
+ * Request: 724871
+ * For IDE plugins (like the eclipse plugin) it would be useful to have
+ * a programmatic access to the first sentence of the TokenType constants,
+ * so they can use them in their configuration gui.
+ * @author Pavel Bludov
+ */
+public final class JavadocPropertiesGenerator {
+
+    /**
+     * The command line option to specify the output file.
+     */
+    private static final String OPTION_DEST_FILE = "destfile";
+
+    /**
+     * The width of the CLI help option.
+     */
+    private static final int HELP_WIDTH = 100;
+
+    /**
+     * This regexp is used to extract the first sentence from the text.
+     */
+    private static final Pattern END_OF_SENTENCE_PATTERN = Pattern.compile("(.*?[.?!])(\\s|$)");
+
+    /**
+     * Don't create instance of this class, use the {@link #main(String[])} method instead.
+     */
+    private JavadocPropertiesGenerator() {
+    }
+
+    /**
+     * TokenTypes.properties generator entry point.
+     * @param args the command line arguments
+     * @throws CheckstyleException if parser or lexer failed or if there is an IO problem
+     * @throws ParseException if the command line can not be passed
+     **/
+    public static void main(String... args)
+            throws CheckstyleException, ParseException {
+        final CommandLine commandLine = parseCli(args);
+        if (commandLine.getArgList().size() == 1) {
+            try {
+                writePropertiesFile(commandLine.getArgList().get(0),
+                    commandLine.getOptionValue(OPTION_DEST_FILE));
+            }
+            catch (IOException ex) {
+                throw new CheckstyleException("Failed to write properties file", ex);
```

Comment:
You do not print in exception message any context information, it is sign of improper place to process exception.

Reply:
done.

Reply:
I still do not see see context variables printed in exception message. No additional context - no need to wrap exception.
Please print "commandLine.getArgList().get(0)".